File: typescript/80.ts
function removeDuplicates(nums: number[]): number {
  let n = 1
  let t = 1
  for (let i = 0; i < nums.length - 1; i++) {
    if (nums[i] != nums[i + 1]) {
      nums[n] = nums[i + 1]
      n += 1
      t = 1
    } else {
      t++
      if (t == 2) {
        nums[n] = nums[i + 1]
        n += 1
      }
    }
  }
  return n
}

File: ocaml/80.ml
let rec f = function
  | a :: (b :: c as t) -> if a = b then f t else a :: f t
  | x -> x

let rec f2 nums =
  match nums with
  | a :: (b :: _ as t) -> if a = b then f2 t else a :: f2 t
  | x -> x

let rec f3 nums =
  match nums with
  | h :: t -> if t != [] && h = List.hd t then f3 t else h :: f3 t
  | x -> x
